board/stm32h7/board.h

```diff
@@ -0,0 +1,36 @@
+// ///////////////////////////////////////////////////////////// //
+// Hardware abstraction layer for all different supported boards //
+// ///////////////////////////////////////////////////////////// //
+#include "boards/board_declarations.h"
+
+// ///// Board definition and detection ///// //
+#include "drivers/harness.h"
+#include "drivers/fan.h"
+#include "stm32h7/llfan.h"
+#include "stm32h7/llrtc.h"
+#include "drivers/rtc.h"
+#include "boards/red.h"
+
+uint8_t board_id(void) {
+  uint8_t bit0 = detect_with_pull(GPIOF, 7, PULL_UP);
+  uint8_t bit1 = (detect_with_pull(GPIOF, 8, PULL_UP) << 1U);
+  uint8_t bit2 = (detect_with_pull(GPIOF, 9, PULL_UP) << 2U);
+  uint8_t bit3 = (detect_with_pull(GPIOF, 10, PULL_UP) << 3U);
+  return (bit3 | bit2 | bit1 | bit0);
+}
+
+void detect_board_type(void) {
+  if(board_id() == 0U){
+    hw_type = HW_TYPE_RED_PANDA;
+    current_board = &board_red;
+  } else {
+    hw_type = HW_TYPE_UNKNOWN;
+    puts("Hardware type is UNKNOWN!\n");
+  }
+}
+
+bool has_external_debug_serial = 0;
+void detect_external_debug_serial(void) {
+  // detect if external serial debugging is present
+  has_external_debug_serial = detect_with_pull(GPIOA, 3, PULL_DOWN);
```

Comment:
is this a thing on the red panda as well? never really knew what this was for

Reply:
It is in the main.c and uart driver(so in common codebase). It is not a thing in red panda yet, but we have those pins routed for USART.